go/chat/sbs_test.go

```diff
@@ -204,76 +220,123 @@ func runChatSBSScenario(t *testing.T, testCase sbsTestCase) {
 
 			mustPostLocalEphemeralForTest(t, ctc, users[0], ncres.Conv.Info,
 				chat1.NewMessageBodyWithText(chat1.MessageText{
-					Body: "HI",
+					Body: "Hi from user 0 (after resolution)",
 				}), ephemeralLifetime)
 			consumeNewMsgRemote(t, listener0, chat1.MessageType_TEXT)
 			consumeNewMsgRemote(t, listener1, chat1.MessageType_TEXT)
 
 			mustPostLocalEphemeralForTest(t, ctc, users[1], ncres.Conv.Info,
 				chat1.NewMessageBodyWithText(chat1.MessageText{
-					Body: "HI",
+					Body: "Hi from user 1 (after resolution)",
 				}), ephemeralLifetime)
 			consumeNewMsgRemote(t, listener0, chat1.MessageType_TEXT)
 			consumeNewMsgRemote(t, listener1, chat1.MessageType_TEXT)
-			verifyThread := func(user *kbtest.FakeUser) {
-				tvres, err := ctc.as(t, user).chatLocalHandler().GetThreadLocal(ctx, chat1.GetThreadLocalArg{
-					ConversationID: ncres.Conv.GetConvID(),
-					Query: &chat1.GetThreadQuery{
-						MessageTypes: []chat1.MessageType{chat1.MessageType_TEXT},
-					},
-				})
-				require.NoError(t, err)
-				require.Equal(t, 3, len(tvres.Thread.Messages))
 
-				for _, msg := range tvres.Thread.Messages {
+			verifyThread := func(user *kbtest.FakeUser, local bool) {
+				var messages []chat1.MessageUnboxed
+				if local {
+					tvres, err := ctc.as(t, user).chatLocalHandler().GetThreadLocal(ctx, chat1.GetThreadLocalArg{
+						ConversationID: ncres.Conv.GetConvID(),
+						Query: &chat1.GetThreadQuery{
+							MessageTypes: []chat1.MessageType{chat1.MessageType_TEXT},
+						},
+					})
+					require.NoError(t, err)
+					messages = tvres.Thread.Messages
+				} else {
+					tc := ctc.world.Tcs[user.Username]
+					ctx := ctc.as(t, user).startCtx
+					// Nuke DB so we don't just pull cached messages.
+					_, err = tc.G.LocalDb.Nuke()
+					require.NoError(t, err)
+					_, err := tc.G.LocalChatDb.Nuke()
+					require.NoError(t, err)
+					tv, err := tc.Context().ConvSource.Pull(
```

Comment:
I wasn't able to completely bypass all caches and simulate fresh load of this thread. Even without the patch to `chat/teams.go` this part of the test would still occasionally successfully load the thread.